libsolidity/ast/Types.cpp

```diff
@@ -739,18 +739,18 @@ bool RationalNumberType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 {
 	if (_convertTo.category() == Category::Integer)
 	{
-		auto targetType = dynamic_cast<IntegerType const*>(&_convertTo);
```

Comment:
There was no assert for this being null, but since the category is checked, better to ref cast.